internal/timeline/tweener.go

```diff
@@ -0,0 +1,308 @@
+package timeline
+
+import (
+	"log"
+	"math"
+
+	"github.com/goplus/spx/internal/math32"
+)
+
+var debugTweener bool = true
+
+type TweenerMode int
+
+const (
+	CUR_TO_END TweenerMode = iota
+	START_TO_END
+	START_TO_CUR
+)
+
+type ITweener interface {
+	ITimeline
+	GetTweener() *Tweener
+	// Rewind()
+}
+
+type Tweener struct {
+	Timeline
+	mode     TweenerMode
+	getter   func() interface{}
+	setter   func(interface{})
+	startVal interface{}
+	curVal   interface{} // only for START_TO_END mode
+	endVal   interface{}
+	duration TIME
+}
+
+func (t *Tweener) GetTweener() *Tweener {
+	return t
+}
+
+func (t *Tweener) Init1(getter func() interface{}, setter func(interface{}), endVal interface{}, duration TIME) *Tweener {
```

Comment:
If you want function overloading, you should name the following functions in this way."
 Init1   => Init__0
 Init2   => Init__1
 Init3   => Init__2

Reply:
Overloading format will be used.